docker/indexer/stages/preparation/preparation.go

```diff
@@ -43,7 +43,7 @@ import (
 const workers = 5
 
 var (
-	genericVersionRE = regexp.MustCompile(`(\d+\.\d+\.?\d?-?(rc|alpha|beta|preview)?\d?)`)
+	genericVersionRE = regexp.MustCompile(`(?i)(\d+|(?<![a-z])(?:rc|alpha|beta|preview)\d*)`)
```

Comment:
Does ?<! work in Go's re syntax?

Reply:
It does not, just replaced it with an alternative, which should mostly have the same behavior. The main difference is:
```
1_5_0-6alpha4
```
will get `1-5-0-6-alpha4` in python, but `1-5-0-6-4` in golang.

Not sure how to resolve those without getting too complicated or using an alternative regex engine.